src/verifiers/issuerIdentity/didIdentityProof/didIdentityProof.test.ts

```diff
@@ -1,33 +1,105 @@
   describe("v2", () => {
-    it("should return false for documents not using DID as top level identifier", () => {});
-    it("should return true for documents where any issuer is using the `DID` identity proof", () => {});
-    it("should return true for documents where any issuer is using the `DNS-CONTRACT` identity proof", () => {});
-  });
-  describe("v3", () => {
-    it("should return false for documents not using DID as top level identifier", () => {});
-    it("should return true for documents where any issuer is using the `DID` identity proof", () => {});
-    it("should return true for documents where any issuer is using the `DNS-CONTRACT` identity proof", () => {});
+    it("should return false for documents not using DID as top level identifier", () => {
+      expect(OpenAttestationDidSignedDidIdentityProof.test(documentRopstenValidWithDocumentStore, options)).toBe(false);
+    });
+    it("should return true for documents where any issuer is using the `DID` identity proof", () => {
+      expect(OpenAttestationDidSignedDidIdentityProof.test(documentDidSigned, options)).toBe(true);
+    });
+    it("should return false for documents where any issuer is using the `DNS-DID` identity proof", () => {
+      expect(OpenAttestationDidSignedDidIdentityProof.test(documentDnsDidSigned, options)).toBe(false);
+    });
   });
 });
 
 describe("verify", () => {
   describe("v2", () => {
-    it("should pass for documents using `DID` and did signature is correct", () => {});
-    it("should pass for documents using `DID-CONTRACT` and documentStore is correctly signed", () => {});
-    it("should fail for documents using `DID` and did signature is not correct", () => {});
-    it("should fail for documents using `DID-CONTRACT` and documentStore is incorrectly signed", () => {});
-  });
-  describe("v3", () => {
-    it("should pass for documents using `DID` and did signature is correct", () => {});
-    it("should pass for documents using `DID-CONTRACT` and documentStore is correctly signed", () => {});
-    it("should fail for documents using `DID` and did signature is not correct", () => {});
-    it("should fail for documents using `DID-CONTRACT` and documentStore is incorrectly signed", () => {});
+    it("should pass for documents using `DID` and did signature is correct", async () => {
+      const verificationFragment = await OpenAttestationDidSignedDidIdentityProof.verify(documentDidSigned, options);
+      expect(verificationFragment).toMatchInlineSnapshot(`
+        Object {
+          "data": Array [
+            Object {
+              "did": "did:ethr:0xE712878f6E8d5d4F9e87E10DA604F9cB564C9a89",
+              "status": "VALID",
+            },
+          ],
+          "name": "OpenAttestationDidSignedDidIdentityProof",
+          "status": "VALID",
+          "type": "ISSUER_IDENTITY",
+        }
+      `);
+    });
+    it("should fail for documents using `DID` and did signature is not correct", async () => {
```

Comment:
I disagree with this test, the signature verification is already made by the document status check, and there is no extra need to do this for the identity (unlike for dns)

Reply:
I disagree with your disagreement. 

There is the need to check for the claim "I am did:xxx:yyy", you perform the check by checking that "did:xxx:yyy" indeed has signed on it. While you perform the same check on the document status, its for different purposes. You will need the module for both `DOCUMENT_STATUS` and `ISSUER_IDENTITY` anyway to pass the final verification.

Reply:
I disagree about your disagreement about my disagreement.

On a more serious note the more we move on, the less I'm convinced that supporting ALL did as identity verifier is not the way to go:
- the exact same work is done twice for 2 different. Just this sounds silly.
- can any DID be used to "verify" identity ? Maybe. Should we support any DID to verify identity ? No
- we have DID-DNS which can provide better identity verification than this. We have been opinionated from the beginning why not continuing and only supporting DNS-DID ?
- we can provide signing with any DID and consumer can provide their own identity verification if they want to
- we can in the future allow some did to perform identity verification (sp/cp ? github?) but I dont think we should allow ALL did

Reply:
Well the problem is that people may not recognize DNS as the best form of identity. 

Using DID, we are telling them, use whatever. If the issuer issues and the recipient accepts, we don't really care. What we want to do is provide the assurance that it has been signed correctly. If you look at the challenges DKIM faces(https://tools.ietf.org/html/rfc6376#section-8), we have them too with DNS-TXT & DNS-DID. 

Like it or not, `example.com`, `Raymond Yeh`, `199201624D`, `S9876543D`, `did:ethr:0xab..cd`, `did:github:nebulis` are all identifiers. Some of them you understand, some you might understand if you google for it, some you might understand and someone else doesnt, and some you can verify. We should be interested in the class of what is verifiable.

Reply:
Ultimately, the user should be educated that they decide whether to accept or not accept documents from a DID, not you. The base layer should be agnostic to this because it is either:
1. Application layer choice (government document verifier might only allow a subset of identifiers)
2. User layer choice (individual choose which identity scheme and identity to believe in)

Reply:
If the base layer could verify it, it should, and the application or user can choose how it wants to deal with it.

Reply:
If you talk about executing the same thing twice, you might want to revisit https://github.com/Open-Attestation/adr/blob/master/verifier.md on the concept of the 3 classes of verification, `DOCUMENT_INTEGRITY`, `DOCUMENT_STATUS` and `ISSUER_IDENTITY`. 

I'm not sure if you disagree at the 3 classes level or at the implementation level.

Reply:
> Well the problem is that people may not recognize DNS as the best form of identity.
> Using DID, we are telling them, use whatever. If the issuer issues and the recipient accepts, we don't really care. What we want to do is provide the assurance that it has been signed correctly. If you look at the challenges DKIM faces(tools.ietf.org/html/rfc6376#section-8), we have them too with DNS-TXT & DNS-DID.
> Like it or not, example.com, Raymond Yeh, 199201624D, S9876543D, did:ethr:0xab..cd, did:github:nebulis are all identifiers. Some of them you understand, some you might understand if you google for it, some you might understand and someone else doesnt, and some you can verify. We should be interested in the class of what is verifiable.

I agree and they can implement their own identity verification if needed we dont have to cover everything as long as we dont think it's not necessary.
I indeed don't like that `example.com` is thought as identity equally as `did:ethr:0xab..cd` or even `did:github:nebulis` (no I dont like much github as identity too :))
I dont think it's incompatible to allow signing with DID and only recognizing DNS-DID by default.

> Ultimately, the user should be educated that they decide whether to accept or not accept documents from a DID, not you. The base layer should be agnostic to this because it is either:
> Application layer choice (government document verifier might only allow a subset of identifiers)
> User layer choice (individual choose which identity scheme and identity to believe in)
By experience you should know that user will just use whatever works. This strengthen my opinion on we shouldn't allow everything by default. They want to allow specific DID ? Go ahead write your own verifier. They want a verifier to be added into OA ? can discuss

> If you talk about executing the same thing twice, you might want to revisit https://github.com/Open-Attestation/adr/blob/master/verifier.md on the concept of the 3 classes of verification, DOCUMENT_INTEGRITY, DOCUMENT_STATUS and ISSUER_IDENTITY.
> I'm not sure if you disagree at the 3 classes level or at the implementation level.
I'm fine with the 3 classes but DID verification does not so any identity verification imho.

Reply:
So to document our discussion:

L: you used did:xyz ? sorry oa doesn tknow write your own verifier

R: 

the beauty of did is that:
1. You use did:xyz:abc, I dont know what is that but it's giving me a public key that I can check
2. The document has been signed by did:xyz, it can be proven that did:xyz signed on this document
3. Trust this document if:
- a. You trust the xyz verification method (xyz could be corppass)
- b. You trust the abc entity

How is that different from DNS:
1. You use xyz.com, I dont know who is that but it's got a binding to a document store that I can check
2. The document has been issued on the document store. It can be proven that xyz.com signed on this document
3. Trust this document if:
- a. You trust DNS
- b. You trust xyz.com as a company or whatever

L: to me the problem is more to not allow any did by default because it will just cascade everywhere

(by default being the most important word)

R: the whole purpose of did is so to get connected to all 🙈

"interoperability"

L: if it's fully open and we fucked on sth we will be fucked up forever

Reply:
actually I think at the OA level there's only two watertight outcomes:
 - either we completely drop the pretense that we are doing any sort of identity assurance and thus -> we provide a means to retrieve identity metadata but we don't say its an assurance we provide -> issuer_identity verifier should always be skipped unless provided by the consumer (but we can provide a DNS one for them to compose into their solution)
 - we are gatekeepers - this will exclude the acceptance of arbitrary DIDs as issuer identities (its a valid argument to make  that this also excludes arbitrary domain names as issuer identities)

any other combination is leaky and will result in "but then x is not better than y because both of them are the same"